part-spec/definitions.json

```diff
@@ -1,250 +1,339 @@
-                    "componentUnit": ["minValue", "maxValue", "value"]
-                }
-            }
-        },
-        "pinFunction": {
-            "type": "object",
-            "properties": {
-                "function" : {
-                    "description": "function of a pin",
-                    "examples": [
-                        "i2c",
-                        "spi",
-                        "gpio"
-                    ],
-                    "type": "string"
-                },
-                "direction" : {
-                    "description": "direction and electrical characteristics of a pin's function",
-                    "examples": [
-                        "in (input)",
-                        "out (output, push-pull)",
-                        "od (output, open-drain)",
-                        "iod (input or output, open-drain)",
-                        "inout (input or output, push-pull)",
-                        "analog",
-                        "power",
-                        "ground"
-                    ],
-                    "enum": [
-                        "in",
-                        "out",
-                        "od",
-                        "iod",
-                        "inout",
-                        "analog",
-                        "power",
-                        "ground"
-                    ],
-                    "type": "string"
-                }
-            }
-        },
-        "unit": {
-            "type": "object",
-            "properties": {
-                "siUnit": {
-                    "description": "name of SI unit of measure",
-                    "enum": [
-                        "volts",
-                        "amps",
-                        "ohms",
-                        "coulomb",
-                        "henries",
-                        "farads"
-                    ],
-                    "type": "string"
-                },
-                "value": {
-                    "description": "unit quantity corresponding to unit text - example 40mV would have a value of 40",
-                    "type": "number"
-                },
-                "unitText": {
-                    "description": "human readable text describing value",
-                    "type": "string"
-                },
-                "unitFactor": {
-                    "description": "multiplier on the value to achieve the SI unit - example for 40mV the unitFactor would be 0.001",
-                    "type": "number"
-                }
-            }
+          "description": "list of external component structures recommended to be attached to a pin",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/externalComponents"
+          }
         }
+      }
+    },
+    "externalComponents": {
+      "type": "object",
+      "required": [
+        "type",
+        "configuration"
+      ],
+      "properties": {
+        "type": {
```

Comment:
Looking at this again, "type" here can be confusing with  "type" we use for data type. Change this property to "componentType"?